guides/bulk.md

```diff
@@ -0,0 +1,209 @@
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Id(1)
+        .Document(new { Title = "The Lion King", Year = 1994 })
+    ));
+```
+
+We will use this format for the rest of the examples in this guide.
+
+### Creating multiple documents
+
+Similarly, instead of calling the `create` method for each document, you can use the `bulk` API to create multiple documents in a single request. The following code creates three documents in the `movies` index and one in the `books` index:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 2", Year = 2030 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 3", Year = 2031 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 4", Year = 2049 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    ));
+```
+
+Note that we specified only the `_index` for the last document in the request body. This is because the `bulk` method accepts an `index` parameter that specifies the default `_index` for all bulk operations in the request body. Moreover, we omit the `_id` for each document and let OpenSearch generate them for us in this example, just like we can with the `create` method.
+
+### Updating multiple documents
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Update<object>(i => i
+        .Index(movies)
+        .Id(1)
+        .Document(new { Year = 1992 })
+    )
+    .Update<object>(i => i
+        .Index(movies)
+        .Id(2)
+        .Document(new { Year = 2018 })
+    ));
+```
+
+Note that the updated data is specified in the `doc` field of the `data` object.
+
+### Deleting multiple documents
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Delete<object>(i => i
+        .Index(movies)
+        .Id(1)
+    )
+    .Delete<object>(i => i
+        .Index(movies)
+        .Id(2)
+    ));
+```
+
+### Mix and match operations
+
+You can mix and match the different operations in a single request. The following code creates two documents, updates one document, and deletes another document:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 5", Year = 2050 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 6", Year = 2051 })
+    )
+    .Update<object>(i => i
+        .Index(movies)
+        .Id(3)
+        .Document(new { Year = 2052 })
+    )
+    .Delete<object>(i => i
+        .Index(movies)
+        .Id(4)
+    ));
```

Comment:
The Index operations should be Create operations, the cleaner version of this is like so:
```suggestion
var response = await client.BulkAsync(b => b
    .Index(movies)
    .CreateMany(new[]
    {
        new { Title = "Beauty and the Beast 5", Year = 2050 },
        new { Title = "Beauty and the Beast 6", Year = 2051 }
    })
    .Update<object>(i => i
        .Id(3)
        .Doc(new { Year = 2052 })
    )
    .Delete<object>(i => i.Id(4)));
```